expo/expo
# feature(jest-expo): load environment variables when defined through dotenv files

# Why

Fixes ENG-11243

This adds support for dotenv files to `jest-expo`, through `@expo/env`.

# How

- Load `@expo/env` from `expo -> @expo/cli -> @expo/env`
- Load `@expo/config/paths` from `expo -> @expo/config/paths`
- Initialize all environment variables in native (non-web) environments
- Initialize only `EXPO_PUBLIC_` environment variables in web (non-native) environments

# Test Plan

Setup test project:

- `$ bun create expo ./test-jest --template tabs`
- `$ bun jest` (run once for the snapshots)
- Add both the `@expo/env` and `jest-expo` code from this PR through:
  - `$ npm pack` in `@expo/env` and `jest-expo` folders
  - Move tarballs to project
  - Update **package.json** `{ "jest-expo": "./tar.tgz" }` (also for `@expo/env`)
- Add `.env` with:
  ```
  HELLO_WORLD=working
  EXPO_PUBLIC_HELLO_WORLD=working
  ```

Setup test:

- Add test code with `expect(process.env.HELLO_WORLD).toBe('working');`
- `$ bun jest` should work
- Change preset to `jest-expo/web`
- `$ bun jest`, only `EXPO_PUBLIC_HELLO_WORLD` should work


# Checklist
